pinot-core/src/main/java/org/apache/pinot/core/segment/processing/framework/SegmentConfig.java

```diff
@@ -72,15 +81,21 @@ public Builder setSegmentNamePrefix(String segmentNamePrefix) {
       return this;
     }
 
+    public Builder setSegmentNamePostfix(String segmentNamePostfix) {
+      _segmentNamePostfix = segmentNamePostfix;
+      return this;
+    }
+
     public SegmentConfig build() {
       Preconditions.checkState(_maxNumRecordsPerSegment > 0, "Max num records per segment must be > 0");
-      return new SegmentConfig(_maxNumRecordsPerSegment, _segmentNamePrefix);
+      return new SegmentConfig(_maxNumRecordsPerSegment, _segmentNamePrefix, _segmentNamePostfix);
     }
   }
 
   @Override
   public String toString() {
-    return "SegmentConfig{" + "_maxNumRecordsPerSegment=" + _maxNumRecordsPerSegment + ", _segmentNamePrefix='"
-        + _segmentNamePrefix + '\'' + '}';
+    return String
```

Comment:
(minor) The current one is auto-generated. Not sure if you want to keep it auto-generated (simply remove the current one and type `toString()` and IDE will suggest the auto-generated one)